data/src/main/java/com/linkedin/data/schema/resolver/FileDataSchemaResolver.java

```diff
@@ -241,7 +266,7 @@ protected DataSchemaLocation transform(String path)
           StringBuilder builder = new StringBuilder();
           // within a JAR file, files are treated as resources. Thus, we should lookup using the resource separator
           // character, which is '/'
-          builder.append(InternalConstants.PEGASUS_DIR_IN_JAR)
+          builder.append(_schemasDirectoryName.getName())
```

Comment:
For "/extensions", we have to support loading from both /pegasus and /extensions paths. Because extension schema includes base schema in /pegasus. If we do not load both paths, the extension schemas can not be resolved.

Reply:
Is it actually both , not "either ... or.."?

Reply:
@BrianPin  For /extensions directory contains ER extensions schemas, which includes base schema which is under /pegasus directory, if we don't loading both, extension schemas cannot be resolved by missing base schemas. 
If directorName is "pegasus", we only load pegasus, if it is "extensions", we have to load both.

Reply:
clear

Reply:
One question for now,
The parser usually takes a resolver path, and the resolver takes the resolver path for finding schema files to resolve.

But for this extension schema, should we have two resolver path?

If we just have one resolver path, we seem can't resolver schemas in the other directory.

Reply:
Added Extensions multi resolver. Please take a look

Reply:
Hi, few comments (Mostly as a question type of comments)
1. The file `data/src/main/java/com/linkedin/data/schema/resolver/ClasspathResourceDataSchemaResolver.java` and this `data/src/main/java/com/linkedin/data/schema/resolver/ExtensionsDataSchemaResolver.java` two resolvers are a bit confusing in terms of when should we use which.  I feel I can have either one to work with extensions.

2. The addition of `_schemasDirectoryName` in `data/src/main/java/com/linkedin/data/schema/resolver/FileDataSchemaResolver.java` seems to me that the addition's purpose is for us to tell the schema prefix (or directories), this thing itself seems more relavant to `ClasspathResourceDataSchemaResolver` or `ExtensionsDataSchemaResolver`, The FileDataSchemaResolver seems can remove this `_schemasDirectoryName` member?

More like two questions above, we can chat offline

Reply:
Answered offline. Also, feel free ping directly for any doubts.
ClasspathResource is for schema already loaded using classLoader, whereas FileDataResolver loads schema from any file directory.